Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArrayFeatureProcessor.cpp

```diff
@@ -537,6 +549,102 @@ namespace AZ
             }
         }
 
+        bool DecalTextureArrayFeatureProcessor::HasCulling(const RPI::ViewPtr& view) const
+        {
+            // First find if there's a render pipeline that is using the View
+            for (const auto& renderPipeline : GetParentScene()->GetRenderPipelines())
+            {
+                if (!renderPipeline->NeedsRender())
+                {
+                    continue;
+                }
+
+                for (const auto& pipelineView : renderPipeline->GetPipelineViews())
+                {
+                    auto findIt = AZStd::find(pipelineView.second.m_views.begin(), pipelineView.second.m_views.end(), view);
+                    if (findIt != pipelineView.second.m_views.end())
+                    {
+                        // Now check if the render pipeline has a pass that is already doing gpu culling for decals
+                        if (renderPipeline->FindFirstPass(AZ::Name("LightCullingPass")))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        void DecalTextureArrayFeatureProcessor::CullDecals(const RPI::ViewPtr& view)
+        {
+            if (!AZ::RHI::CheckBitsAll(view->GetUsageFlags(), RPI::View::UsageFlags::UsageCamera) ||
+                HasCulling(view))
```

Comment:
Rather than checking `HasCulling` I think we should consider a compile time constant #define like `AZ_TRAIT_USE_CPU_DECAL_CULLING` and setting that to true on mobile platforms, and in the project if desired.

Reply:
A #define will not work as this logic is based of render pipeline used which maybe different for different devices. A cvar can be considered here as a debugging tool to capture how much perf is gained by doing cpu culling although we would never want to disable this optimization as an override so not sure how valuable this cvar will be. I would be ok not adding this cvar at all.